indexer/ai/src/main/java/com/google/cloudsearch/LocalRepository.java

```diff
@@ -0,0 +1,183 @@
+package com.google.cloudsearch;
+
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.services.cloudsearch.v1.model.Item;
+import com.google.cloudsearch.ai.AISkill;
+import com.google.cloudsearch.ai.AISkillSet;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.primitives.Longs;
+import com.google.enterprise.cloudsearch.sdk.CheckpointCloseableIterable;
+import com.google.enterprise.cloudsearch.sdk.CheckpointCloseableIterableImpl;
+import com.google.enterprise.cloudsearch.sdk.RepositoryException;
+import com.google.enterprise.cloudsearch.sdk.config.ConfigValue;
+import com.google.enterprise.cloudsearch.sdk.config.Configuration;
+import com.google.enterprise.cloudsearch.sdk.indexing.Acl;
+import com.google.enterprise.cloudsearch.sdk.indexing.IndexingItemBuilder;
+import com.google.enterprise.cloudsearch.sdk.indexing.IndexingService;
+import com.google.enterprise.cloudsearch.sdk.indexing.template.ApiOperation;
+import com.google.enterprise.cloudsearch.sdk.indexing.template.Repository;
+import com.google.enterprise.cloudsearch.sdk.indexing.template.RepositoryContext;
+import com.google.enterprise.cloudsearch.sdk.indexing.template.RepositoryDoc;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+import java.util.stream.IntStream;
+
+public class LocalRepository implements Repository {
+
+    private static Logger log = Logger.getLogger(LocalRepository.class.getName());
+    private AISkillSet skillSet;
+    private List<String> allFileNames = new ArrayList<String>();
+
+
+    @Override
+    public void init(RepositoryContext repositoryContext) throws RepositoryException {
+        //Read and store file paths from the resource folder
+
+        ConfigValue<String> folderName = Configuration.getValue("resources.names",null,Configuration.STRING_PARSER);
+        log.info(folderName.get());
+        File[] files = getAllFiles(folderName.get());
+        for (File f : files) {
+            log.info(f.getPath());
+            allFileNames.add(f.getPath());
+        }
+        //Parse the JSON
+        ConfigValue<String> AISkillConfig = Configuration.getValue("enrichment.config", null, Configuration.STRING_PARSER);
```

Comment:
As discussed in previous PR, lets pass the config to AISkillSet class. All json parsing should be taken care by the AISkillSet class only